domain/schema/model/sql/0019-unit.sql

```diff
@@ -52,31 +52,30 @@ ON unit (application_uuid);
 CREATE INDEX idx_unit_net_node
 ON unit (net_node_uuid);
 
-CREATE TABLE unit_platform (
+-- unit_principal table is a table which is used to store the.
+-- principal units for subordinate units.
+CREATE TABLE unit_principal (
     unit_uuid TEXT NOT NULL PRIMARY KEY,
-    os_id TEXT,
-    channel TEXT,
-    architecture_id TEXT,
-    CONSTRAINT fk_unit_platform_unit
+    principal_uuid TEXT NOT NULL,
+    CONSTRAINT fk_unit_principal_unit
     FOREIGN KEY (unit_uuid)
     REFERENCES unit (uuid),
-    CONSTRAINT fk_unit_platform_os
-    FOREIGN KEY (os_id)
-    REFERENCES os (id),
-    CONSTRAINT fk_unit_platform_architecture
-    FOREIGN KEY (architecture_id)
-    REFERENCES architecture (id)
+    CONSTRAINT fk_unit_principal_principal
+    FOREIGN KEY (principal_uuid)
+    REFERENCES unit (uuid)
 );
 
-CREATE TABLE unit_tool (
+CREATE TABLE unit_agent (
     unit_uuid TEXT NOT NULL,
     url TEXT NOT NULL,
     version_major INT NOT NULL,
     version_minor INT NOT NULL,
     version_tag TEXT,
     version_patch INT NOT NULL,
     version_build INT,
-    CONSTRAINT fk_unit_tool_unit
+    sha256 TEXT NOT NULL,
```

Comment:
This should be a link table to a hash table. This is the standard in every other table. It allows to migrate away from sha256. See:

https://github.com/juju/juju/blob/3808fb2e0203a6e602e9083bacd19c02a4137198/domain/schema/model/sql/0015-charm.sql#L125-L141

Reply:
This was copied across from what was already done for machine. Anyway, I've added an extra column to record the hash kind. I really don't see the need for a link table here. The unit_agent and machine_agent tables are already adjust to the core unit and machine entities.

Why do we have **3** hash kind look up tables. Surely just `hash_kind` will do.
```
INSERT INTO ssh_fingerprint_hash_algorithm VALUES
(0, 'md5'),
(1, 'sha256');
```
```
INSERT INTO hash_kind VALUES
(0, 'sha256');
```
```
INSERT INTO object_store_metadata_hash_type VALUES
(0, 'none'),
(1, 'sha512-384');
```

Reply:
We should unify. I agree, the hash_kind is spot on.